components/tidb_query_expr/src/impl_time.rs

```diff
@@ -150,6 +150,40 @@ pub fn to_seconds(ctx: &mut EvalContext, t: &DateTime) -> Result<Option<Int>> {
     Ok(Some(t.second_number()))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn add_datetime_and_duration(
+    ctx: &mut EvalContext,
+    datetime: &DateTime,
+    duration: &Duration,
+) -> Result<Option<Time>> {
+    let mut res = match datetime.checked_add(ctx, *duration) {
+        Some(res) => res,
+        None => return Ok(None),
+    };
+    if res.set_time_type(TimeType::DateTime).is_err() {
```

Comment:
Is this `set_time_type` check necessary here? @skyzh

Reply:
I think it's better to add this check.

Reply:
I think retain this check is okay.